File: src/main/java/dev/interfiber/karpet/server/recipes/MinecraftRecipe.java
package dev.interfiber.karpet.server.recipes;

import net.minestom.server.entity.Player;
import net.minestom.server.item.ItemStack;
import net.minestom.server.item.Material;
import net.minestom.server.network.packet.server.play.DeclareRecipesPacket.Ingredient;

import java.util.List;

public class MinecraftRecipe {
    public String RecipeID;
    public List<ItemStack> Items; // 6x6 square grid, for anything blank mark it as Material.AIR
    public ItemStack Result;

    public boolean CanCraft(Player player) {
        return false;
    }
    public boolean CanCraftInPortableCrafting() { return false; }
}

File: src/main/kotlin/dev/interfiber/karpet/server/recipes/MinecraftRecipe.kt
package dev.interfiber.karpet.server.recipes

import net.minestom.server.item.ItemStack

/**
 * Represent a craftable recipe
 * @author Interfiber
 */
open class MinecraftRecipe {
    /**
     * A unique id to represent the recipe
     * @author Interfiber
     */
    var recipeID: String? = null
    /**
     * List of items used to craft the item in the portable table
     * @author Interfiber
     */
    var portableItems
            : List<ItemStack>? = null
    /**
     * List of items used to craft the item in the crafting table
     * @author Interfiber
     */
    var items
        : List<ItemStack>? = null
    /**
     * The output of the crafting recipe
     * @author Interfiber
     */
    var result: ItemStack? = null
//    fun CanCraft(player: Player?): Boolean {
//        return false
//    }

    open fun canCraftInPortableCrafting(): Boolean {
        return false
    }
}